x-pack/legacy/plugins/infra/public/utils/formatters/data.ts

```diff
@@ -62,12 +62,19 @@ const BASES = {
   [InfraWaffleMapDataFormat.abbreviatedNumber]: 1000,
 };
 
-export const createDataFormatter = (format: InfraWaffleMapDataFormat) => (val: number) => {
+const BIT_BASES = [
+  InfraWaffleMapDataFormat.bitsBinaryJEDEC,
+  InfraWaffleMapDataFormat.bitsBinaryIEC,
+  InfraWaffleMapDataFormat.bitsDecimal,
+];
+
+export const createDataFormatter = (format: InfraWaffleMapDataFormat) => (bytes: number) => {
   const labels = LABELS[format];
   const base = BASES[format];
-  const power = Math.min(Math.floor(Math.log(Math.abs(val)) / Math.log(base)), labels.length - 1);
+  const value = BIT_BASES.includes(format) ? bytes * 8 : bytes;
+  const power = Math.min(Math.floor(Math.log(Math.abs(value)) / Math.log(base)), labels.length - 1);
```

Comment:
What does this calculation do, and why? Could you add a comment?

Reply:
As I read it, it determines the `power` in `base**power` given a `base` to decide which abbreviated unit to use (kilo, mega, ..., kibi, mebi, ...) for a `value`.

Reply:
I added a comment. I'm using an [exponential equation](http://www.montereyinstitute.org/courses/DevelopmentalMath/COURSE_TEXT2_RESOURCE/U18_L4_T1_text_final.html) to figure out the power used for the label. If the power is greater then the max label then we just use the max label.